networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/topics/GossipTopics.java

```
@@ -70,4 +74,17 @@ public static String getSyncCommitteeSubnetTopic(
 
     return topics;
   }
+
+  public static Optional<Bytes4> extractForkDigest(final String topic) {
+    // Fork digest starts after domain prefix + slash separators
+    final int beginIndex = DOMAIN_PREFIX.length() + 2;
+    final int endIndex = topic.indexOf("/", beginIndex);
+    final String forkDigest = topic.substring(beginIndex, endIndex);
+
+    try {
+      return Optional.of(Bytes4.fromHexString(forkDigest));
+    } catch (IllegalArgumentException e) {
+      return Optional.empty();
+    }
```

Since we wind up throwing an exception if this returns empty anyway, should we just let this throw  the `IllegalArgumentException` and not return Optional?